### DISCUSSION TITLE
Mobile navbar unexpectedly sliding in / out on first click when enabling "delay JS execution" with WP Rocket

### DISCUSSION THREAD

**Discussion by Scott716** [holger1411/understrap#2226] [Q&A]
What file contains the JS that Bootstrap needs for offcanvas navigation, so we can exclude it?

We run forceofnatureclean.com with Understrap, a child theme, and (importantly) **WP Rocket** for faster page loads.

But when we set WP Rocket to "delay JS execution" the mobile navbar unexpected slides in / out when you first click it.

Here is a video from our test environment

https://share.zight.com/jkujng4g

Enabling 'defer JS execution' would almost halve page load times, so we want it enabled.

We can name specific JS files to be excluded from 'defer JS execution' but we've already excluded every one included with Understrap and the problem persists.

This is the setting in WP Rocket:
![Screen Shot 2024-12-19 at 2 52 07 PM](https://github.com/user-attachments/assets/094bfac6-21f0-498b-b0f6-a884ce785fda)

**Comment by IanDelMar** [comment]
Understrap bundles all the Bootstrap scripts in [theme.min.js](https://github.com/understrap/understrap/blob/main/js/theme.min.js). What and where you are loading scripts in your child theme is known only to you.
